### PR TITLE
Upgrade Node to 20

### PR DESCRIPTION
**Pull request by Fabian-Fynn**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -34,31 +34,28 @@
...
+        "lint-staged": "^15.2.7",
         "prettier": "^2.0.0",
-        "typescript": "^4.5.5",
-        "yarn-run-all": "^3.0.0"
-    },
-    "engines": {
-        "node": ">=14"
+        "typescript": "^5.5.3",
+        "yarn-run-all": "^3.1.1"
     }
```
[the remark this answers is not in the document]
Comment (johnnyomair) [reply package.json:60]: We should keep the engines field.

**Review by johnnyomair — COMMENTED** [review]
General feedback: Updating all dependencies and Node could have been two separate PRs, as they don't rely on each other. But it's okay for the size of this PR.

File: package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (Fabian-Fynn) [reply package.json:63]: yes we should keep it.

**Review by kaufmo — DISMISSED** [review]

File: .changeset/angry-cooks-clean.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"@comet/dev-process-manager": major
```
Comment (nsams) [inline .changeset/angry-cooks-clean.md:2]:
```suggestion
"@comet/dev-process-manager": minor
```

Comment (johnnyomair) [reply .changeset/angry-cooks-clean.md:2]: If we don't enforce the version then we should keep engines at >= 14 and remove the changeset altogether

**Review by johnnyomair — APPROVED** [review]

**Review by nsams — APPROVED** [review]

**Review by kaufmo — APPROVED** [review]

File: .changeset/angry-cooks-clean.md (hunk 2 above)
Comment (kaufmo) [reply .changeset/angry-cooks-clean.md]: @johnnyomair FYI we do not enforce the engines but our subpackages do for example command v12 enforces nodejs >= 18 now

**Pull request merged**